app/services/migration/migrator.rb

```diff
@@ -39,25 +41,69 @@ def check_environment!
 
     def run_migration
       migrate_lead_providers
+      migrate_cohorts
+      migrate_statements
     end
 
-    def migrate_lead_providers
-      data_migration = DataMigration.find_by(model: :lead_provider)
-
-      data_migration.update!(started_at: Time.zone.now, total_count: ecf_npq_lead_providers.count)
+    def migrate(items, model)
+      data_migration = DataMigration.find_by(model:)
+      data_migration.update!(started_at: Time.zone.now, total_count: items.count)
 
-      ecf_npq_lead_providers.find_each do |ecf_npq_lead_provider|
+      items.each do |item|
         data_migration.increment!(:processed_count)
-        npq_lead_provider = LeadProvider.find_by(ecf_id: ecf_npq_lead_provider.id)
-
-        data_migration.increment!(:failure_count) unless npq_lead_provider
+        result = yield(item)
+        data_migration.increment!(:failure_count) unless result
       end
 
       data_migration.update!(completed_at: Time.zone.now)
     end
 
+    def migrate_lead_providers
+      migrate(ecf_npq_lead_providers, :lead_provider) do |ecf_npq_lead_provider|
+        LeadProvider.find_by(ecf_id: ecf_npq_lead_provider.id)
+      end
+    end
+
     def ecf_npq_lead_providers
       @ecf_npq_lead_providers ||= Ecf::NpqLeadProvider.all
     end
+
+    def ecf_cohorts
+      @ecf_cohorts ||= Ecf::Cohort.where(start_year: 2021..)
+    end
+
+    def migrate_cohorts
+      migrate(ecf_cohorts, :cohort) do |ecf_cohort|
+        cohort = Cohort.find_or_initialize_by(start_year: ecf_cohort.start_year)
+        result = cohort.update(registration_start_date: ecf_cohort.registration_start_date)
+        result
+      end
+    end
+
+    def ecf_statements
+      @ecf_statements ||= Ecf::Finance::Statement.all
+    end
+
+    def migrate_statements
+      migrate(ecf_statements, :statement) do |ecf_statement|
+        statement = Statement.find_or_initialize_by(month: Date::MONTHNAMES.find_index(ecf_statement.name.split[0]), year: ecf_statement.name.split[1])
+
+        result = statement.update(
+          deadline_date: ecf_statement.deadline_date,
+          payment_date: ecf_statement.payment_date,
+          output_fee: ecf_statement.output_fee,
+          cohort: Cohort.find_by(start_year: ecf_cohorts.find_by(id: ecf_statement.cohort_id).start_year),
+          lead_provider: LeadProvider.find_by(ecf_id: ecf_npq_lead_providers.find_by(cpd_lead_provider_id: ecf_statement.cpd_lead_provider_id).id),
```

Comment:
```suggestion
          lead_provider: LeadProvider.find_by(ecf_id: ecf_statement.cpd_lead_provider.npq_lead_provider.id),
```
since we import both tables maybe this is simpler?